src/server/src/main/java/io/cassandrareaper/service/Heart.java

```diff
@@ -93,80 +99,97 @@ public void close() {
+                                          "metrics",
+                                          req.getCluster(),
+                                          req.getHostAddress())) {
+
+                                    try (JmxProxy nodeProxy =
+                                        context.jmxConnectionFactory.connect(
+                                            req.getHostAddress(), jmxTimeoutSeconds)) {
+
+                                      storage.storeNodeMetrics(
+                                          NodeMetrics.builder()
+                                              .withHostAddress(req.getHostAddress())
+                                              .withDatacenter(nodeProxy.getDataCenter())
+                                              .withPendingCompactions(
+                                                  nodeProxy.getPendingCompactions())
+                                              .withHasRepairRunning(nodeProxy.isRepairRunning())
+                                              .withActiveAnticompactions(0) // for future use
+                                              .build());
+
+                                      ((IDistributedStorage) context.storage)
+                                          .removeNodeMetricsRequests(
+                                              req.getRunId(), req.getHostAddress());
+
+                                      LOG.info(
+                                          "Responded to metric request for node {}",
+                                          req.getHostAddress());
+                                    } catch (ReaperException
+                                        | RuntimeException
+                                        | InterruptedException ex) {
+                                      LOG.debug(
+                                          "failed seed connection in cluster " + req.getCluster(),
+                                          ex);
+                                    } catch (AttributeNotFoundException
+                                        | MBeanException
+                                        | ReflectionException e) {
+                                      LOG.warn(
+                                          "failed querying JMX MBean for metrics on node {} of cluster {}",
+                                          req.getHostAddress(),
+                                          req.getCluster(),
+                                          e);
+                                    }
+                                  }
+                                });
+                      })
+                  .get();
+            } catch (ExecutionException | InterruptedException | RuntimeException ex) {
+              LOG.warn("failed updateAllReachableNodeMetrics submission", ex);
+            } finally {
+              assert updatingNodeMetrics.get();
+              updatingNodeMetrics.set(false);
+            }
+          });
```

Comment:
👍  this was along the lines that i'd envisioned :-)

i'm not too sure why the nesting it deeper than before, would have expected the opposite. will take a look…